python/nav/ipdevpoll/shadows/__init__.py

```diff
@@ -142,6 +145,7 @@ def prepare(self, containers):
         self._fix_binary_garbage()
         self._fix_missing_name()
         self._resolve_duplicate_names()
+        self.is_new = not self.get_existing_model()
```

Comment:
I might actually advocate for `is_new` as a new *property* of `Shadow` objects (also, it's not considered good practice to create new instance attributes outside of the `__init__` method).

Reply:
Unsure how to do this, as the self.get_existing_model() check has to be done before saving to be able to tell if it existed before. so something like 

```
@propery
def is_new(self):
    return not self.get_existing_model
```

wouldnt really work if its called during cleanup. Alternatively it could be set in the Shadow __init__, but then a db query would be made for every container instance regardless of if you care about it being new or not. I've moved the is_new definitions to overridden __init__ for the relevant shadow classes for the time being

Reply:
> Unsure how to do this, as the self.get_existing_model() check has to be done before saving to be able to tell if it existed before. so something like
> 
> ```
> @propery
> def is_new(self):
>     return not self.get_existing_model
> ```
> 
> wouldnt really work if its called during cleanup. 

True, although it depends on how you expect to use it.

After saving an object, it isn't "new" any longer. But if the intent is to know it "was new" before the save ran, then a different logic is needed. At the time of `__init__`, you also don't really know whether the object is new or old - strictly speaking, if the object is entirely empty at the time of `__init__`, `is_new` would evaluate to `True`, until you added some attribute values that suddenly changes the evaluation to `False`.

> Alternatively it could be set in the Shadow **init**, but then a db query would be made for every container instance regardless of if you care about it being new or not. I've moved the is_new definitions to overridden **init** for the relevant shadow classes for the time being

IIRC, your intent with this was actually know, during cleanup, if the object `was_new` before the save operation. So maybe `is_new` isn't the right name in this case?